spring/src/main/java/io/cloudevents/spring/messaging/MapContextMessageReader.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cloudevents.spring.messaging;
+
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+import io.cloudevents.CloudEventData;
+import io.cloudevents.SpecVersion;
+import io.cloudevents.core.message.impl.BaseGenericBinaryMessageReaderImpl;
+
+import static io.cloudevents.spring.messaging.CloudEventsHeaders.CE_PREFIX;
+import static org.springframework.messaging.MessageHeaders.CONTENT_TYPE;
+
+/**
+ * Utility for converting maps (message headers) to `CloudEvent` contexts.
+ * 
+ * @author Dave Syer
+ *
+ */
+class MapContextMessageReader extends BaseGenericBinaryMessageReaderImpl<String, Object> {
```

Comment:
Cab we have another name here? Like `SpringMessageReader` or something like that? `MapContext` is confusing to me because it's not only about a map and about context.

Reply:
I think bringing 'Spring' into the name would be even more confusing, since there is really nothing specific to Spring here other then `MessageHeaders.CONTENT_TYPE` which could be removed in favor of some local constant. Perhaps `KeyValueMessageReader`?

Reply:
Yeah perhaps SpringMessageReader doesn't sound right, `KeyValueMessageReader` doesn't sound right too because here you're handling the body too... Not sure what's the right name tbh, but `MapContextMessageReader` is for sure wrong... Maybe we can live with just `MessageReader`?

Reply:
I don't care. It was private originally, so the name didn't matter.

Reply:
`MessageReader` makes no sense to me though - it duplicates the interface that it implements.